commit 49de4585a650c5e70cace733cb466fd007526f80 (origin/master, origin/HEAD)
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu May 28 10:49:51 2020 +0200

    libdiskfs,libnetfs: add missing io_pathconf cases

    * libdiskfs/io-pathconf.c (diskfs_S_io_pathconf): For
    _PC_REC_MAX_XFER_SIZE, _PC_REC_INCR_XFER_SIZE, _PC_SYMLINK_MAX, return
    undefined. For _PC_2_SYMLINKS, return 1. For _PC_REC_MIN_XFER_SIZE,
    _PC_REC_XFER_ALIGN, _PC_ALLOC_SIZE_MIN, return page size.
    * libnetfs/io-pathconf.c (netfs_S_io_pathconf): Likewise.

Index: hurd-debian/libdiskfs/io-pathconf.c
===================================================================
--- hurd-debian.orig/libdiskfs/io-pathconf.c
+++ hurd-debian/libdiskfs/io-pathconf.c
@@ -42,6 +42,9 @@ diskfs_S_io_pathconf (struct protid *cre
     case _PC_VDISABLE:
     case _PC_SOCK_MAXBUF:
     case _PC_PATH_MAX:
+    case _PC_REC_MAX_XFER_SIZE:
+    case _PC_REC_INCR_XFER_SIZE:
+    case _PC_SYMLINK_MAX:
       *value = -1;
       break;
 
@@ -55,6 +58,7 @@ diskfs_S_io_pathconf (struct protid *cre
       break;
 
     case _PC_NO_TRUNC:		/* enforced in diskfs_lookup */
+    case _PC_2_SYMLINKS:
       *value = 1; /* diskfs_name_max >= 0; */ /* see above */
       break;
 
@@ -72,6 +76,12 @@ diskfs_S_io_pathconf (struct protid *cre
       *value = 32;
       break;
 
+    case _PC_REC_MIN_XFER_SIZE:
+    case _PC_REC_XFER_ALIGN:
+    case _PC_ALLOC_SIZE_MIN:
+      *value = vm_page_size;
+      break;
+
     default:
       return EINVAL;
     }
Index: hurd-debian/libnetfs/io-pathconf.c
===================================================================
--- hurd-debian.orig/libnetfs/io-pathconf.c
+++ hurd-debian/libnetfs/io-pathconf.c
@@ -40,6 +40,9 @@ netfs_S_io_pathconf (struct protid *user
     case _PC_VDISABLE:
     case _PC_SOCK_MAXBUF:
     case _PC_PATH_MAX:
+    case _PC_REC_MAX_XFER_SIZE:
+    case _PC_REC_INCR_XFER_SIZE:
+    case _PC_SYMLINK_MAX:
       *value = -1;
       break;
       
@@ -49,6 +52,7 @@ netfs_S_io_pathconf (struct protid *user
 
     case _PC_CHOWN_RESTRICTED:
     case _PC_NO_TRUNC:		/* look at string_t trunc behavior in MiG */
+    case _PC_2_SYMLINKS:
       *value = 1;
       break;
       
@@ -62,6 +66,12 @@ netfs_S_io_pathconf (struct protid *user
       *value = 32;
       break;
 
+    case _PC_REC_MIN_XFER_SIZE:
+    case _PC_REC_XFER_ALIGN:
+    case _PC_ALLOC_SIZE_MIN:
+      *value = vm_page_size;
+      break;
+
     default:
       return EINVAL;
     }
